--- /etc/apparmor.d/usr.bin.thunderbird.orig	2022-01-23 20:16:44.866098074 +0100
+++ /etc/apparmor.d/usr.bin.thunderbird	2022-01-23 20:21:33.280378016 +0100
@@ -55,6 +55,8 @@
   # With older GLib we might still be on the fallback code path
   # (remove this after Debian 11 and Ubuntu 20.04)
   /usr/lib/@{multiarch}/glib-[0-9]*/gio-launch-desktop ix,
+  # Allow opening links in Tor Browser
+  owner @{HOME}/.local/share/torbrowser/tbb/x86_64/tor-browser_en-US/start-tor-browser.desktop Cx -> torbrowser_tor,
 
   # For Xubuntu to launch the browser
   /usr/bin/exo-open ixr,
