--- /usr/share/doc/fzf/examples/key-bindings.zsh.orig	2022-08-01 11:52:28.639735804 +0200
+++ /usr/share/doc/fzf/examples/key-bindings.zsh	2022-08-01 12:01:40.371223642 +0200
@@ -101,9 +101,15 @@
     FZF_DEFAULT_OPTS="--height ${FZF_TMUX_HEIGHT:-40%} $FZF_DEFAULT_OPTS -n2..,.. --tiebreak=index --bind=ctrl-r:toggle-sort,ctrl-z:ignore $FZF_CTRL_R_OPTS --query=${(qqq)LBUFFER} +m" $(__fzfcmd)) )
   local ret=$?
   if [ -n "$selected" ]; then
+    local edit=0
+    if [[ $selected[1] = left ]] || [[ $selected[1] = right ]]; then
+        edit=1
+        shift selected
+    fi
     num=$selected[1]
     if [ -n "$num" ]; then
       zle vi-fetch-history -n $num
+      [[ $edit = 1 ]] || zle accept-line
     fi
   fi
   zle reset-prompt
